### PR TITLE
WIP add dependency injection and mocks to tests

### PR DESCRIPTION
**Pull request by apontini**
This PR aims at adding dependency injection support and mocks to tests.
This is now implemented with Guice and Mockito.
We might also consider [Koin](https://github.com/InsertKoinIO/koin) to perform depencendy injection, which is build natively for Kotlin.
Before merging, we need to discuss which one to use

### CODE REVIEW

File: src/main/kotlin/it/polpetta/utils/JenkinsSession.kt [hunk 1]
```diff
@@ -26,8 +26,8 @@ import it.polpetta.config.Resources
 import java.io.FileNotFoundException
 import java.nio.file.Path
 
-object Jenkins {
-    val session: JenkinsClient?
+class JenkinsSession {
```
Comment (Polpetta) [inline src/main/kotlin/it/polpetta/utils/JenkinsSession.kt:29]: Can't we use objects anymore with Guice?

**Review by Polpetta — APPROVED** [review]
I mean it seems good to me. Do you think Koin will be better in the long run?
The fact is that this project has to use less human resources as possible, and we already know how Guice works... but I trust you here, I don't really know what to do! :sweat_smile:

File: src/main/kotlin/it/polpetta/utils/JenkinsSession.kt (hunk 1 above)
Comment (apontini) [reply src/main/kotlin/it/polpetta/utils/JenkinsSession.kt:29]: No we actually can, i didn't remember to change it back to object.
On a side note, objects cannot have a constructor, therefore no dependencies can be injected, which might make them less usefult

**Comment by apontini** [comment]
I was googling some informations about Koin and then I found [this article on Medium](https://medium.com/@stoltmanjan/what-is-up-with-koin-719d1248562c) which explains some pros and cons about it. I didn't really like the cons, I feel that going with Guice will be the right thing

**Pull request merged**